src/sentry/sentry_apps/api/endpoints/installation_details.py

```diff
@@ -39,26 +43,35 @@ def get(self, request: Request, installation) -> Response:
         )
 
     def delete(self, request: Request, installation) -> Response:
-        installation = SentryAppInstallation.objects.get(id=installation.id)
+        sentry_app_installation: SentryAppInstallation = SentryAppInstallation.objects.get(
+            id=installation.id
+        )
         with transaction.atomic(using=router.db_for_write(SentryAppInstallation)):
             try:
-                InstallationNotifier.run(install=installation, user=request.user, action="deleted")
+                assert isinstance(
+                    request.user, (User, RpcUser)
+                ), "User must be a user or rpcuser to delete installation"
```

Comment:
We could use likely use `assert request.user.is_authenticated` here instead of explict type checks.

Reply:
huh interesting, so it appears if we accept `User | RpcUser`, `is_authenticated` will be ok but if we're expecting `RpcUser` then it's an issue.